### ISSUE TITLE
Jenkins build does not run Portal Integration Tests

### ISSUE THREAD

**Issue by maysunfaisal** [eclipse/codewind#1611]
**Codewind version:**
**OS:**

**Che version:**
**IDE extension version:**
**IDE version:**
**Kubernetes cluster:**

**Description:**
I have noticed on Dec 20 that Jenkins PR builds do not run Portal's integration tests.

I am not sure how long this is broken, it means we've been checking in PRs without running it against the integration tests.

From one of the PRs

```
13:58:49  +++   TURBINE UNIT TESTS COMPLETED SUCCESSFULLY   +++
[Pipeline] }
[Pipeline] // withDockerRegistry
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // timeout
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Run Codewind test suite)
[Pipeline] timeout
13:58:49  Timeout set to expire in 2 hr 0 min
[Pipeline] {
[Pipeline] withEnv
[Pipeline] {
[Pipeline] withDockerRegistry
13:58:50  $ docker login -u codewindbot -p ******** https://index.docker.io/v1/
13:58:50  WARNING! Using --password via the CLI is insecure. Use --password-stdin.
13:58:51  WARNING! Your password will be stored unencrypted in /home/root/workspace/Codewind_codewind_PR-1588@tmp/13de9e5f-fb04-4501-bf42-4c0821a89533/config.json.
13:58:51  Configure a credential helper to remove this warning. See
13:58:51  https://docs.docker.com/engine/reference/commandline/login/#credentials-store
13:58:51  
13:58:51  Login Succeeded
[Pipeline] {
[Pipeline] sh
13:58:53  Starting tests for Eclipse Codewind ...
13:58:53  
13:58:53  
13:58:53  Platform: x86_64 
13:58:53    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
13:58:53                                   Dload  Upload   Total   Spent    Left  Speed
13:58:53  
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 13226  100 13226    0     0   516k      0 --:--:-- --:--:-- --:--:--  516k
13:58:53  => nvm is already installed in /root/.nvm, trying to update using git
13:58:53  
=> => Compressing and cleaning up git repository
13:58:53  
13:58:53  => nvm source string already in /root/.bashrc
13:58:53  => bash_completion source string already in /root/.bashrc
13:58:55  => Close and reopen your terminal to start using nvm or run the following to use it now:
13:58:55  
13:58:55  export NVM_DIR="$HOME/.nvm"
13:58:55  [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
13:58:55  [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
13:58:56  v10.18.0 is already installed.
13:58:56  Now using node v10.18.0 (npm v6.13.4)
13:58:56    % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
13:58:56                                   Dload  Upload   Total   Spent    Left  Speed
13:58:57  
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   617    0   617    0     0   2448      0 --:--:-- --:--:-- --:--:--  2448
13:58:57  
  1 10.3M    1  202k    0     0   251k      0  0:00:42 --:--:--  0:00:42  251k
100 10.3M  100 10.3M    0     0  7389k      0  0:00:01  0:00:01 --:--:-- 16.1M
13:59:08  
13:59:08  > nodemon@1.19.4 postinstall /home/root/workspace/Codewind_codewind_PR-1588/src/pfe/portal/node_modules/nodemon
13:59:08  > node bin/postinstall || exit 0
13:59:08  
13:59:08  npm WARN codewind-portal@1.0.0 No description
13:59:08  npm WARN codewind-portal@1.0.0 No repository field.
13:59:08  npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/fsevents):
13:59:08  npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
13:59:08  
13:59:08  added 731 packages from 491 contributors and audited 3785 packages in 9.585s
13:59:08  
13:59:08  9 packages are looking for funding
13:59:08    run `npm fund` for details
13:59:08  
13:59:08  found 0 vulnerabilities
13:59:08  
13:59:09  
13:59:09  > codewind-portal@1.0.0 eslint /home/root/workspace/Codewind_codewind_PR-1588/src/pfe/portal
13:59:09  > eslint .
13:59:09  
13:59:11  
13:59:11  /home/root/workspace/Codewind_codewind_PR-1588/src/pfe/portal/modules/ExtensionList.js
13:59:11     42:33  warning  Expected 'this' to be used by class method 'installBuiltInExtensions'  class-methods-use-this
13:59:11     66:15  warning  Unexpected `await` inside a loop                                       no-await-in-loop
13:59:11     67:13  warning  Unexpected `await` inside a loop                                       no-await-in-loop
13:59:11     71:13  warning  Unexpected `await` inside a loop                                       no-await-in-loop
13:59:11     80:11  warning  Unexpected `await` inside a loop                                       no-await-in-loop
13:59:11     97:24  warning  Unexpected `await` inside a loop                                       no-await-in-loop
13:59:11    101:13  warning  Unexpected `await` inside a loop                                       no-await-in-loop
13:59:11    104:15  warning  Unexpected `await` inside a loop                                       no-await-in-loop
13:59:11  
13:59:11  /home/root/workspace/Codewind_codewind_PR-1588/src/pfe/portal/modules/FileWatcher.js
13:59:11    210:19  warning  Expected 'this' to be used by class method 'logFWReturnedMsg'  class-methods-use-this
13:59:11  
13:59:11  /home/root/workspace/Codewind_codewind_PR-1588/src/pfe/portal/modules/LoadRunner.js
13:59:11    252:1   warning  Expected indentation of 8 spaces but found 6  indent
13:59:11    254:31  warning  Unexpected `await` inside a loop              no-await-in-loop
13:59:11  
13:59:11  /home/root/workspace/Codewind_codewind_PR-1588/src/pfe/portal/modules/Project.js
13:59:11    256:9   warning  Unexpected `await` inside a loop  no-await-in-loop
13:59:11    326:27  warning  Unexpected `await` inside a loop  no-await-in-loop
13:59:11    665:20  warning  Unexpected `await` inside a loop  no-await-in-loop
13:59:11    670:9   warning  Unexpected `await` inside a loop  no-await-in-loop
13:59:11  
13:59:11  /home/root/workspace/Codewind_codewind_PR-1588/src/pfe/portal/modules/User.js
13:59:11    187:7   warning  Unexpected `await` inside a loop  no-await-in-loop
13:59:11    238:11  warning  Unexpected `await` inside a loop  no-await-in-loop
13:59:11  
13:59:11  /home/root/workspace/Codewind_codewind_PR-1588/src/pfe/portal/modules/utils/dockerFunctions.js
13:59:11    59:3  warning  Async function 'inspect' expected no return value  consistent-return
13:59:11  
13:59:11  /home/root/workspace/Codewind_codewind_PR-1588/src/pfe/portal/routes/projects/fileChanges.route.js
13:59:11    74:50  warning  Async function has no 'await' expression  require-await
13:59:11  
13:59:11  /home/root/workspace/Codewind_codewind_PR-1588/src/pfe/portal/routes/projectTypes.route.js
13:59:11    98:23  warning  Unexpected `await` inside a loop  no-await-in-loop
13:59:11  
13:59:11  ✖ 20 problems (0 errors, 20 warnings)
13:59:11    0 errors and 1 warning potentially fixable with the `--fix` option.
13:59:11  
13:59:11  
13:59:11  
13:59:11  Creating codewind-workspace
13:59:11  The HOST_MAVEN_OPTS variable is not set. Defaulting to a blank string.
13:59:11  Creating network "codewind_network" with the default driver
13:59:11  Creating volume "codewind_cw-workspace" with default driver
13:59:11  Building codewind-gatekeeper
13:59:11  Step 1/12 : FROM node:12-alpine
13:59:11   ---> e12611f14560
13:59:11  Step 2/12 : LABEL org.opencontainers.image.title="Codewind-Gatekeeper" org.opencontainers.image.description="Codewind Gatekeeper"       org.opencontainers.image.url="https://codewind.dev/"       org.opencontainers.image.source="https://github.com/eclipse/codewind"
13:59:11   ---> Using cache
13:59:11   ---> 22db1b7de3fc
13:59:11  Step 3/12 : EXPOSE 9096 9496
13:59:11   ---> Using cache
13:59:11   ---> 0aec560d5aaa
13:59:11  Step 4/12 : RUN apk add --no-cache --update openssl
13:59:11   ---> Using cache
13:59:11   ---> 4c60314ec741
13:59:11  Step 5/12 : COPY LICENSE NOTICE.md ./
13:59:13   ---> 84c290631144
13:59:13  Step 6/12 : WORKDIR /usr/src/app
13:59:13   ---> Running in a213e67b6413
13:59:13  Removing intermediate container a213e67b6413
13:59:13   ---> 087ab0ccd6ed
13:59:13  Step 7/12 : COPY . .
13:59:13   ---> 43fd17d524b9
13:59:13  Step 8/12 : ARG IMAGE_BUILD_TIME
13:59:14   ---> Running in 7a30d02b986b
13:59:14  Removing intermediate container 7a30d02b986b
13:59:14   ---> 05eec2040c54
13:59:14  Step 9/12 : ENV IMAGE_BUILD_TIME=${IMAGE_BUILD_TIME}
13:59:14   ---> Running in d07649bacf25
13:59:14  Removing intermediate container d07649bacf25
13:59:14   ---> 33581de92457
13:59:14  Step 10/12 : RUN npm ci --production
13:59:14   ---> Running in 62269b6192dc
13:59:17  added 126 packages in 1.753s
13:59:18  Removing intermediate container 62269b6192dc
13:59:18   ---> 12ec814f5db8
13:59:18  Step 11/12 : USER 1000
13:59:19   ---> Running in 290ac146353b
13:59:19  Removing intermediate container 290ac146353b
13:59:19   ---> 10bcc020bd86
13:59:19  Step 12/12 : CMD [ "npm", "start" ]
13:59:19   ---> Running in ed9cd24c7ee6
13:59:19  Removing intermediate container ed9cd24c7ee6
13:59:19   ---> 18e62431d432
13:59:19  Successfully built 18e62431d432
13:59:19  Successfully tagged codewind-gatekeeper-amd64:latest
13:59:19  Image for service codewind-gatekeeper was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
13:59:19  Building codewind-keycloak
13:59:19  Step 1/5 : FROM jboss/keycloak:8.0.0
13:59:19   ---> edebe94965de
13:59:19  Step 2/5 : LABEL org.opencontainers.image.title="Codewind-Keycloak" org.opencontainers.image.description="Codewind Keycloak"       org.opencontainers.image.url="https://codewind.dev/"       org.opencontainers.image.source="https://github.com/eclipse/codewind"
13:59:19   ---> Using cache
13:59:19   ---> 273c2fc56c0f
13:59:19  Step 3/5 : COPY LICENSE NOTICE.md ./
13:59:19   ---> 19c55c025811
13:59:19  Step 4/5 : COPY ./theme/  /opt/jboss/keycloak/themes/
13:59:19   ---> a0e1227230c7
13:59:19  Step 5/5 : EXPOSE 8080 8443
13:59:19   ---> Running in 9037276cfa75
13:59:19  Removing intermediate container 9037276cfa75
13:59:19   ---> 2aa3c7bd4aa8
13:59:19  Successfully built 2aa3c7bd4aa8
13:59:19  Successfully tagged codewind-keycloak-amd64:latest
13:59:19  Image for service codewind-keycloak was built because it did not already exist. To rebuild this image you must use `docker-compose build` or `docker-compose up --build`.
13:59:19  Building codewind-performance
13:59:20  Cannot locate specified Dockerfile: Dockerfile
13:59:20  
13:59:20  
13:59:20  FAILED TO START CONTAINERS 
[Pipeline] }
[Pipeline] // withDockerRegistry
[Pipeline] }
[Pipeline] // withEnv
[Pipeline] }
[Pipeline] // timeout
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Publish Docker images)
Stage "Publish Docker images" skipped due to when conditional
[Pipeline] }
[Pipeline] // stage
[Pipeline] stage
[Pipeline] { (Declarative: Post Actions)
[Pipeline] sh
13:59:22  
13:59:22  ********** codewind-pfe logs **********
13:59:22  
13:59:22  Error: No such container: codewind-pfe
13:59:22  
13:59:22  Stopping and removing Codewind Docker containers.
13:59:22  
13:59:22  ERROR: THERE ARE NO CONTAINERS TO REMOVE
13:59:22  
13:59:22  ERROR: THERE ARE NO CONTAINERS TO REMOVE 
13:59:22  
13:59:22  Removing docker network
13:59:22  codewind_network
13:59:22  
13:59:22  Successfully removed docker network 

```

**Comment by james-wallis** [comment]
Looks like when the containers fail to build the Jenkinsfile is exiting with a code of 0. 
Think the Fix would be to change the line https://github.com/eclipse/codewind/blob/master/Jenkinsfile#L216 should be changed to be `exit 1;`.

As a further enhancement it would be nice if the `eslint` and `unittest` scripts ran before we built and ran the containers.

**Comment by james-wallis** [comment]
This has been fixed in #1632 , @maysunfaisal can this be closed?
